spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -645,12 +668,13 @@ public void bindPubSubProducer(String name, MessageChannel moduleOutputChannel,
 			Properties properties) {
 		validateProducerProperties(name, properties, SUPPORTED_PUBSUB_PRODUCER_PROPERTIES);
 		RabbitPropertiesAccessor accessor = new RabbitPropertiesAccessor(properties);
-		String exchangeName = applyPrefix(accessor.getPrefix(this.defaultPrefix), applyPubSub(name));
+		String exchangeName = applyPrefix(accessor.getPrefix(this.defaultPrefix), name);
 		declareExchangeIfNotPresent(new FanoutExchange(exchangeName));
```

Comment:
Wouldn't you want this to be a TopicExchange as well, instead of a FanoutExchange?